app/operations/iteration_operations.rb

```diff
@@ -12,16 +12,29 @@ def initialize(start_date:, end_date:, project:)
 
     def run
       {
-        stories: stories
+        stories: past_iteration.stories
       }
     end
 
     private
 
+    attr_reader :project, :start_date, :end_date
+
+    def past_iteration
+      Iterations::PastIteration.new(
+        start_date: start_date,
+        end_date: end_date,
+        stories: stories
+      )
+    end
+
     def stories
-      Iterations::PastIteration.new(start_date: @start_date,
-                                    end_date: @end_date,
-                                    project: @project).stories
+      @stories ||= begin
+        project
+          .stories
+          .with_dependencies
+          .where('accepted_at >= ? AND accepted_at <= ?', start_date, end_date)
```

Comment:
Isn't this the same thing as https://github.com/Codeminer42/cm42-central/pull/379/files#diff-0648f597f84a74e685e5eabc0855e13cR31 ?

Reply:
Almost, but for just one past_iteration...

Reply:
@rodrigovirgilio ☝️

Reply:
@rodrigovirgilio Apply it in start_date and end_date methods.

Reply:
@rodrigovirgilio ☝️